File: baseLib/src/main/java/me/ycdev/android/lib/common/utils/StringUtils.java
package me.ycdev.android.lib.common.utils;

import androidx.annotation.NonNull;

@SuppressWarnings({"unused", "WeakerAccess"})
public class StringUtils {
    public static String trimPrefixSpaces(String str) {
        final int N = str.length();
        int index = 0;
        while (index < N && (str.charAt(index) <= '\u0020' || str.charAt(index) == '\u00a0')) {
            index++;
        }
        if (index > 0) {
            return str.substring(index);
        }
        return str;
    }

    public static int parseInt(@NonNull String value, int defValue) {
        try {
            return Integer.parseInt(value);
        } catch (Exception e) {
            return defValue;
        }
    }

    public static long parseLong(@NonNull String value, long defValue) {
        try {
            return Long.parseLong(value);
        } catch (Exception e) {
            return defValue;
        }
    }
}

File: baseLib/src/main/java/me/ycdev/android/lib/common/utils/StringUtils.kt
package me.ycdev.android.lib.common.utils

@Suppress("unused")
object StringUtils {
    fun trimPrefixSpaces(str: String): String {
        val size = str.length
        var index = 0
        while (index < size && (str[index] <= '\u0020' || str[index] == '\u00a0')) {
            index++
        }
        return if (index > 0) {
            str.substring(index)
        } else {
            str
        }
    }

    fun parseInt(value: String, defValue: Int): Int {
        return try {
            value.toInt()
        } catch (e: Exception) {
            defValue
        }
    }

    fun parseLong(value: String, defValue: Long): Long {
        return try {
            value.toLong()
        } catch (e: Exception) {
            defValue
        }
    }
}
